### DISCUSSION TITLE
Take next lines into account when suggesting

### DISCUSSION THREAD

**Discussion by msanchezdev** [community/community#30911] [Copilot Conversations]
Currently Copilot seems to only be suggesting code according to the code before the current line. So it's not able to suggest code properly according to the current context sometimes.

Example:
Even though I have already added the comments for the `help` property it is suggesting me the comment for the `help` property rather than the `build` property. (Probably taking the order into account from another file in the project which has `help` at the top of the `ValidArgs` interface)
![image](https://user-images.githubusercontent.com/35346082/186894222-ead1f283-db5a-4a45-aa23-08285c6229c8.png)

Even after adding more context for Copilot on the other property it still doesn't take it into account. (If I instead document `build`/`b` first and then `help`/`h` it does provide better suggestions)
![image](https://user-images.githubusercontent.com/35346082/186895494-c8d0281f-5bc7-4ea2-8e19-510cb8fe7eaa.png)

So it would be better if copilot could take the whole file as the context and provide suggestions according to the line you are at. 
#7331 (Accept partial suggestions) would be a plus as sometimes copilot suggests the whole function I am currently writing with code that I already have, thus breaking the code by duplicating the whole function.
Example:
![image](https://user-images.githubusercontent.com/35346082/186896291-24c545ea-7ac8-40c9-a816-ee83fbdfd666.png)

**Comment by pedro-pedrosa** [comment]
this is by biggest issue with copilot. I'm assuming this is very difficult to fix due to the nature of the underlying AI model, since it was built specifically to guess what comes next, as a text generator, not as a code generator.

my workaround is to write all code without comments, then start writing the comments at the end of the file. it builds the context nicely for the comments. then I manually move the comments to the right place. multi cursor editing copy-paste helps a lot here since the generated comments reflect the order of variable declarations in code

**Reply by 1nVitr0** [reply]
This is incredibly annoying when documenting methods. If always duplicates some other method's doc code. And then I have to write it myself, which triggers the auto doc comment, which I have to partly remove, which makes my entire file red because of an unclosed comment and then have to press enter for every parameter. It's still faster than writing it completely myself, but it could be so much faster!
